packages/driver/src/util/serialization.ts

```diff
@@ -27,12 +29,86 @@ const isSerializableInCurrentBrowser = (value: any) => {
 }
 
 /**
- * Sanitizes any unserializable values from a object to prep for postMessage serialization
- * @param objectToSanitize Object that might have unserializable properties
- * @returns a copy of this object with all unserializable keys omitted from the object.
+ * Walks the prototype chain and finds any serializable properties that exist on the object or its prototypes.
+ * If the property can be serialized, the property is added to the literal.
+ * This means read-only properties are now read/write on the literal.
  *
- * NOTE: If an object nested inside objectToSanitize contains an unserializable property, the whole object is deemed as unserializable
+ * Please see https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm#things_that_dont_work_with_structured_clone for more details.
+ * @param obj Object that is being converted
+ * @returns a new object void of prototype chain (object literal) with all serializable properties
  */
-export const preprocessObjForSerialization = <T>(objectToSanitize: { [key: string]: any }): T => {
-  return _.pickBy(objectToSanitize, isSerializableInCurrentBrowser) as T
+const convertObjectToSerializableLiteral = (obj): typeof obj => {
+  const allProps: string[] = []
+  let currentObjectRef = obj
+
+  do {
+    const props = Object.getOwnPropertyNames(currentObjectRef)
+
+    props.forEach((prop: string) => {
+      if (!allProps.includes(prop) && isSerializableInCurrentBrowser(currentObjectRef[prop])) {
+        allProps.push(prop)
+      }
+    })
+
+    currentObjectRef = Object.getPrototypeOf(currentObjectRef)
+  } while (currentObjectRef)
+
+  const objectAsLiteral = {}
+
+  allProps.forEach((key) => {
+    objectAsLiteral[key] = obj[key]
+  })
+
+  return objectAsLiteral
+}
+
+/**
+ * Sanitizes any unserializable values to prep for postMessage serialization. All Objects, including Errors, are mapped to an Object literal with
+ * whatever serialization properties they have, including their prototype hierarchy.
+ * This keeps behavior consistent between browsers without having to worry about the inner workings of structuredClone(). For example:
+ *
+ * chromium
+ * new Error('myError') -> Object literal with message key having value 'myError'. Also, other custom properties on the object are omitted and only name, message, and stack are preserved
+ *
+ * For instance:
+ * var a = new Error('myError')
+ * a.foo = 'bar'
+ * var b = structuredClone(a)
+ * b.foo // is undefined
+ *
+ * firefox
+ * structuredClone(new Error('myError')) -> throws error as native error cannot be serialized
+ *
+ * This method takes a similar the 'chromium' approach to structuredClone, except that the prototype chain is walked and ANY serializable value, including getters, is serialized.
```

Comment:
```suggestion
 * This method takes a similar approach as the 'chromium' structuredClone algorithm, except that the prototype chain is walked and ANY serializable value, including getters, is serialized.
```

Reply:
Thank you for finding this. This sentence doesn't make any sense 🥴 .

Reply:
addressed in 4872216566f6ad7b26933bb99def80295f04e236